File: src/api/microApp.ts
import { post } from '@/utils/request'

// ==================== 前台/公开 API ====================

// 获取微应用列表（公开接口）
export function getList<T>(data: MicroApp.GetListRequest) {
  return post<T>({
    url: '/microApp/getList',
    data,
  })
}

// 获取微应用详情（公开接口 - 新增）
export function getInfo<T>(id: number) {
  return post<T>({
    url: '/microApp/getInfo',
    data: { id },
  })
}

// 获取微应用版本列表（只返回审核通过的）
export function getVersionList<T>(data: MicroApp.GetVersionListRequest) {
  return post<T>({
    url: '/microApp/version/getList',
    data,
  })
}

export function getDownloadUrl<T>(microAppId: string, version?: string) {
  return post<T>({
    url: '/microApp/download/getUrl/',
    data: { microAppId, version },
  })
}

File: service/router/micro_app/micro_app.go
package microApp

import (
	"sun-panel/api/api_v1"

	"github.com/gin-gonic/gin"
)

func InitMicroAppRouter(router *gin.RouterGroup) {
	microAppApi := api_v1.ApiGroupApp.MicroAppApi
	downloadApi := api_v1.ApiGroupApp.DownloadApi

	// ==================== 公开接口（微应用） ====================
	router.POST("microApp/getList", microAppApi.GetList)
	router.POST("microApp/getInfo", microAppApi.GetInfo)
	router.POST("microApp/version/getList", microAppApi.GetVersionList)

	router.POST("microApp/download/getUrl", downloadApi.GetUrl) // 获取下载链接

	// ==================== 下载接口（自动统计） ====================
	router.GET("microApp/download/:microAppId", downloadApi.DownloadByVersionOrLatest)          // 下载最新版本
	router.GET("microApp/download/:microAppId/:version", downloadApi.DownloadByVersionOrLatest) // 下载指定版本
}
